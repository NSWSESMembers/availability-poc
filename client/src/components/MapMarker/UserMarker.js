import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';

import styles from './styles';

const UserMarker = ({ name, status, destination, locationTime }) => (
  <View style={styles.container}>
    <View style={styles.bubble}>
      <Text style={[styles.name]}>{name}</Text>
      {destination &&
        <Text style={[styles.destination]}>
          {status.toUpperCase()} {destination.toUpperCase()}
        </Text>}
      <Text style={[styles.locationTime]}>{locationTime}</Text>
    </View>
    <View style={styles.arrowBorder} />
    <View style={styles.arrow} />
  </View>
);

UserMarker.propTypes = {
  name: PropTypes.string.isRequired,
  destination: PropTypes.string,
  status: PropTypes.string,
  locationTime: PropTypes.string,
};

export default UserMarker;
