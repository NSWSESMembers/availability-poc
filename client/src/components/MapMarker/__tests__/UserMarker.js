import React from 'react';
import renderer from 'react-test-renderer';

import UserMarker from '../UserMarker';

test('renders correctly', () => {
  const tree1 = renderer.create(
    <UserMarker
      name="name"
      status="status"
      destination="destination"
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();
});
