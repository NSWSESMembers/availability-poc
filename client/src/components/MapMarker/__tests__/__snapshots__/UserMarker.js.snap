// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "flex-start",
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#5A6DFF",
        "borderColor": "#161AF9",
        "borderRadius": 3,
        "borderWidth": 0.5,
        "flex": 0,
        "opacity": 0.8,
        "padding": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      name
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#FFFFFF",
            "fontSize": 8,
          },
        ]
      }
    >
      STATUS
       
      DESTINATION
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#FFFFFF",
            "fontSize": 6,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderTopColor": "#D23F44",
        "borderWidth": 4,
        "marginTop": -0.5,
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderTopColor": "#FF5A5F",
        "borderWidth": 4,
        "marginTop": -9,
      }
    }
  />
</View>
`;
